File: app/src/main/java/com/jsh/kr/alltest/util/TestTextUtil.java
package com.jsh.kr.alltest.util;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.Random;

public class TestTextUtil {
    public static String makeCurrentTime() {
        SimpleDateFormat format = new SimpleDateFormat("hh:mm:ss", Locale.KOREA);
        return format.format(new Date());
    }

    public static String makeRandomId() {
        SimpleDateFormat format = new SimpleDateFormat("MMddHHmm", Locale.getDefault());
        return format.format(new Date()) + makeRandomNumEng(2);
    }

    public static String makeRandomNumEng(int cnt) {
        // ASCII 0~9(48 ~ 57),  A~Z(65~90), a(97 ~ 122)
        char[] charArray = new char[cnt];

        for(int i = 0; i < cnt ; i++) {
            Random random = new Random();
            int randomNum = random.nextInt(62);

            LogUtil.d("makeRandomNumEng", "cnt : "+cnt + ", randomNum : "+randomNum);
            if (randomNum < 10) {
                LogUtil.d("makeRandomNumEng", "int value : "+(randomNum + 48) + ", char value : "+(char) (randomNum + 48));
                charArray[i] = (char) (randomNum + 48);
            } else if (randomNum < 36) {
                LogUtil.d("makeRandomNumEng", "int value : "+(randomNum - 10 + 65) + ", char value : "+(char) (randomNum - 10 + 65));
                charArray[i] = (char) (randomNum - 10 + 65);
            } else if (randomNum < 62) {
                LogUtil.d("makeRandomNumEng", "int value : "+(randomNum - 36 + 97) + ", char value : "+(char) (randomNum - 36 + 97));
                charArray[i] = (char) (randomNum - 36 + 97);
            }
        }

        return new String(charArray);
    }

}

File: appkt/src/main/java/com/jsh/kr/alltestkt/util/TestTextUtil.kt
package com.jsh.kr.alltestkt.util

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.Random

object TestTextUtil {
    fun makeCurrentTime(): String? {
        val format = SimpleDateFormat("hh:mm:ss", Locale.KOREA)
        return format.format(Date())
    }

    fun makeRandomId(): String? {
        val format = SimpleDateFormat("MMddHHmm", Locale.getDefault())
        return format.format(Date()) + makeRandomNumEng(2)
    }

    fun makeRandomNumEng(cnt: Int): String {
        // ASCII 0~9(48 ~ 57),  A~Z(65~90), a(97 ~ 122)
        val charArray = CharArray(cnt)
        for (i in 0 until cnt) {
            val random = Random()
            val randomNum = random.nextInt(62)
            LogUtil.d(
                "makeRandomNumEng",
                "cnt : $cnt, randomNum : $randomNum"
            )
            if (randomNum < 10) {
                LogUtil.d(
                    "makeRandomNumEng",
                    "int value : " + (randomNum + 48) + ", char value : " + (randomNum + 48).toChar()
                )
                charArray[i] = (randomNum + 48).toChar()
            } else if (randomNum < 36) {
                LogUtil.d(
                    "makeRandomNumEng",
                    "int value : " + (randomNum - 10 + 65) + ", char value : " + (randomNum - 10 + 65).toChar()
                )
                charArray[i] = (randomNum - 10 + 65).toChar()
            } else if (randomNum < 62) {
                LogUtil.d(
                    "makeRandomNumEng",
                    "int value : " + (randomNum - 36 + 97) + ", char value : " + (randomNum - 36 + 97).toChar()
                )
                charArray[i] = (randomNum - 36 + 97).toChar()
            }
        }
        return String(charArray)
    }
}